### DISCUSSION TITLE
Route params in URL not updating

### DISCUSSION THREAD

**Discussion by haztakki** [livewire/livewire#7253] [1. Help]
Hello,

Updating a route param property inside a component in Livewire v3 no longer updates the URL. Is this a bug or a change? I'm not seeing anything about it in the Upgrade Guide. This worked previously in Livewire v2.

Here's a dumbed-down simplified repo to reproduce:

Livewire v2:

`"livewire/livewire": "v2.12.6"`

`alpinejs@3.x.x`

```php
use App\Livewire\View as Directory;

Route::get('/directory/{directoryId}', Directory::class)
    ->name('directory');
``` 

**View.php:**

```
<?php

namespace App\Livewire;

class View extends \Livewire\Component
{
    protected $listeners = [
        'current.directory.updated' => 'setCurrentDirectory',
    ];

    public int $directoryId;

    public function render(): \Illuminate\Contracts\View\View
    {
        return view('view');
    }

    public function setCurrentDirectory(int $id): void
    {
        $this->directoryId = $id;
    }
}
```

**view.blade.php:**

```blade
<div>
    <div
        x-data=" { setCurrentDirectory(id) { Livewire.emit('current.directory.updated', id); } }"
    >
        <div>
            <button @click.prevent="setCurrentDirectory(1)">Set cwd to 1</button>
        </div>

        <div>
            <button @click.prevent="setCurrentDirectory(2)">Set cwd to 2</button>
        </div>
    </div>
</div>
```

This all works fine. Note the URL here.

https://github.com/livewire/livewire/assets/20573863/1ca0775b-6547-4aad-89b5-d23106d22a0c

But in Livewire v3, this behaviour is different. The route param in the URL does not get updated.

`"livewire/livewire": "^3.0" // v3.1.0`

**View.php:**

```php
<?php

namespace App\Livewire;

class View extends \Livewire\Component
{
    public int $directoryId;

    public function render(): \Illuminate\Contracts\View\View
    {
        return view('view');
    }

    #[\Livewire\Attributes\On('current.directory.updated')]
    public function setCurrentDirectory(int $id): void
    {
        $this->directoryId = $id;
    }
}
```

**view.blade.php:**

```blade
<div>
    <div
        x-data=" { setCurrentDirectory(id) { Livewire.dispatch('current.directory.updated', { id: id } ); } }"
    >
        <div>
            <button @click.prevent="setCurrentDirectory(1)">Set cwd to 1</button>
        </div>

        <div>
            <button @click.prevent="setCurrentDirectory(2)">Set cwd to 2</button>
        </div>
    </div>
</div>
```

https://github.com/livewire/livewire/assets/20573863/34756c08-93df-430e-93c9-15afa5aebdb6

I did a search and found the following, but there doesn't appear to be a solution. I'm curious as to why this behaviour was changed.

https://github.com/livewire/livewire/discussions/6939

Thanks.

**Comment by drifteaur** [comment]
I am experiencing this too - changing route params would update the URL in Livewire 2, but doesn't seem to do that in Livewire 3. Also see #6939 - seems to be the same issue.
